behaviour/graql/language/define.feature

```diff
@@ -1018,92 +928,700 @@ Feature: Graql Define Query
       | VDR |
 
 
-  Scenario: define abstract attribute type creates a type
+  Scenario: define attribute type that has itself creates an attribute
     Given graql define
       """
-      define number-of-limbs sub attribute, abstract, value long;
+      define number-of-letters sub attribute, value long, has number-of-letters;
       """
     Given the integrity is validated
     When get answers of graql query
       """
-      match $x type number-of-limbs; get;
+      match $x has number-of-letters; get;
       """
     Then concept identifiers are
-      |     | check | value           |
-      | NOL | label | number-of-limbs |
+      |     | check | value                |
+      | NOL | label | number-of-letters    |
     Then uniquely identify answer concepts
       | x   |
       | NOL |
 
 
-  Scenario: define concrete subtype of abstract attribute creates child of its parent type
+  #########
+  # RULES #
+  #########
+
+  Scenario: define a rule that infers an attribute value creates a rule
     Given graql define
       """
       define
-      number-of-limbs sub attribute, abstract, value long;
-      number-of-legs sub number-of-limbs;
+      nickname sub name;
+      person has nickname;
+      robert-has-nickname-bob sub rule,
+      when {
+        $p isa person, has name "Robert";
+      }, then {
+        $p has nickname "Bob";
+      };
       """
     Given the integrity is validated
-
     When get answers of graql query
       """
-      match $x sub number-of-limbs; get;
+      match $x sub rule; get;
       """
     Then concept identifiers are
-      |     | check | value           |
-      | NOL | label | number-of-limbs |
-      | NLE | label | number-of-legs  |
+      |     | check | value                   |
+      | BOB | label | robert-has-nickname-bob |
+      | RUL | label | rule                    |
     Then uniquely identify answer concepts
       | x   |
-      | NOL |
-      | NLE |
+      | BOB |
+      | RUL |
 
 
-  Scenario: define abstract subtype of abstract attribute creates child of its parent type
+  Scenario: define a rule that infers a relation creates a rule
```

Comment:
i kind of like having the word "valid" in these titles - "creates a valid rule" lets the reader immediately see that the point is here to have a "valid" result :)

Reply:
Yeah I'm no big fan of "creating a rule creates a rule" either.
I think I might change these to the format: `a rule can infer a relation`.
I was kinda obsessed with including the word "define" in every test, but it's in the filename for god's sakes.

Reply:
Changed to `a rule can infer a relation`.